File: selenium-java-ui-demo/src/test/java/com/demo/webdriveruniversity/pages/HomePage.java
package com.demo.webdriveruniversity.pages;

import com.demo.base.Config;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.support.ui.WebDriverWait;

public class HomePage extends WduBasePage {

  private final By heading = By.xpath("//h1[contains(., 'Practise')]");
  private final By contactUsCard = By.id("contact-us");

  public HomePage(WebDriver driver, WebDriverWait wait) {
    super(driver, wait);
  }

  public void goTo() {
    driver.get(Config.get("wdu.baseUrl") + "/index.html");
  }

  public By heading() {
    return heading;
  }

  public void openContactUs() {
    openChallenge(driver.findElement(contactUsCard));
  }
}

File: webdriverio-ui-demo/src/websites/webdriveruniversity/pages/HomePage.ts
import { $ } from '@wdio/globals';
import type { ChainablePromiseElement } from 'webdriverio';
import { WduBasePage } from './WduBasePage.js';

class HomePage extends WduBasePage {
  get heading(): ChainablePromiseElement {
    return $('h1*=Practise');
  }

  get contactUsCard(): ChainablePromiseElement {
    return $('#contact-us');
  }

  async goto(): Promise<void> {
    await browser.url('/index.html');
  }

  async openContactUs(): Promise<void> {
    await this.openChallenge(this.contactUsCard);
  }
}

export default new HomePage();
